packages/graphql-js-operation-payload-generator/src/index.ts

```diff
@@ -227,16 +294,28 @@ function visitDocumentDefinitionNode(
  */
 function reduceToSingleObjectTypeSelection(
   selectionSetNode: SelectionSetNode,
+  explicitTypename: string | undefined,
   schema: GraphQLSchema,
   allDocumentDefinitionNodes: ReadonlyArray<DefinitionNode>,
   typeInfo: TypeInfo
 ): SelectionSetNode {
   let reduceToObjectType: GraphQLObjectType | null = null;
```

Comment:
I'm confused by this code in general, why do we need to keep it as global state? I won't block merge cause it's been there before, but this is weird.

Reply:
Unsure which global state you’re referring to, can you elaborate?